### ISSUE TITLE
Add `horovod.run()` to make horovod notebook friendly

### ISSUE THREAD

**Issue by mengxr** [horovod/horovod#1176]
**Is your feature request related to a problem? Please describe.**

`horovod.spark.run` makes it easy for users to run Horovod on Spark. The interface that takes a train method is quite convenient and fits notebook users well:) However, it only supports Spark. Sometimes, users might want to run distributed training on a multi-GPU node inside a notebook environment, or users might want to trigger an arbitrary job with worker IPs. Right now, they still need to create a Python script and triggered via `horovodrun` command-line.

**Describe the solution you'd like**

I want to have a Python method works similarly to `horovodrun` but takes a python method instead. It will pickle the method to disk and trigger the MPI job using `horovodrun`.

~~~python
import horovod
import horovod.tensorflow as hvd

def train_hvd():
  hvd.init()
  # training code

horovod.run(train_hvd, args, kwargs, num_proc, hosts, ...)
~~~

Users should be able to quickly modify `train_hvd` method and re-run `horoovd.run`. The simplification is that users do not need to manually save the code to disk and call `horovodrun`

@WeichenXu123 will submit a prototype.

cc: @alsrgv

**Comment by alsrgv** [comment]
LGTM, looking forward to the prototype!

**Comment by denfromufa** [comment]
I like this approach, we were just thinking about this last week.

**Comment by tgaddair** [comment]
Fixed by #1192.
